### PR TITLE
DAOS-6086 EC: Do not convert offset from parity

### PR DESCRIPTION
**Pull request by wangdi1**
Do not need convert offset for IOM, if it is from
parity shard.

Add more dfs degrade ec tests.

Some daos log for debugging purpose.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

**Review by liuxuezhao — APPROVED** [review]

File: src/object/cli_shard.c [hunk 1]
```diff
@@ -390,8 +391,9 @@ obj_ec_iom_merge(struct obj_reasb_req *reasb_req, uint32_t tgt_idx,
 	lo = src->iom_recx_lo;
 	end = DAOS_RECX_END(lo);
 	lo.rx_nr = min(end, roundup(lo.rx_idx + 1, cell_rec_nr)) - lo.rx_idx;
```
[the remark this answers is not in the document]
Comment (liuxuezhao) [reply src/object/cli_shard.c]: I refreshed the patch to fix a few other cases, please see the 2nd commit in this PR. thx

**Review by liuxuezhao — DISMISSED** [review]

**Review by liuxuezhao — APPROVED** [review]

File: src/tests/ftest/daos_test/daos_core_test.yaml [hunk 2]
```diff
@@ -144,9 +144,15 @@ daos_tests:
     test_Z:
       daos_test: Z
       test_name: DAOS aggregate ec tests
-    test_X:
+    test_X_1:
       daos_test: X
-      test_name: DAOS degraded ec tests
+      test_name: DAOS degraded ec tests 0-6
+      args: -u subtests="0-6"
+      test_timeout: 900
```
Comment (jolivier23) [inline src/tests/ftest/daos_test/daos_core_test.yaml:151]: is full 900 needed?

Comment (wangdi1) [reply src/tests/ftest/daos_test/daos_core_test.yaml:151]: ah, probably not. I will shrink it if it needs to refresh

File: src/object/cli_shard.c [hunk 3]
```diff
@@ -357,25 +357,26 @@ obj_ec_iom_merge(struct obj_reasb_req *reasb_req, uint32_t shard,
...
 		daos_recx_ep_list_hilo(recov_list, &recov_hi, &recov_lo);
 
+	is_data_shard = (shard % obj_ec_tgt_nr(oca)) < obj_ec_data_tgt_nr(oca);
 	D_SPIN_LOCK(&reasb_req->orr_spin);
 
 	/* merge iom_recx_hi */
 	hi = src->iom_recx_hi;
 	end = DAOS_RECX_END(hi);
-	if (end > 0)
+	if (end > 0 && is_data_shard) {
```
Comment (wangdi1) [inline src/object/cli_shard.c:374]: I thought for parity shard, end is always 0 here

File: src/object/cli_shard.c [hunk 4]
```diff
@@ -390,12 +391,14 @@ obj_ec_iom_merge(struct obj_reasb_req *reasb_req, uint32_t shard,
 	/* merge iom_recx_lo */
 	lo = src->iom_recx_lo;
 	end = DAOS_RECX_END(lo);
-	lo.rx_nr = min(end, roundup(lo.rx_idx + 1, cell_rec_nr)) - lo.rx_idx;
-	if (shard < obj_ec_data_tgt_nr(oca))
+	if (end > 0 && is_data_shard) {
```
Comment (wangdi1) [inline src/object/cli_shard.c:394]: lo is also 0 for parity shard? no?

Comment (wangdi1) [reply src/object/cli_shard.c:394]: Btw: this to and hi merging seems a bit duplicate?

File: src/object/cli_shard.c (hunk 3 above)
Comment (liuxuezhao) [reply src/object/cli_shard.c:374]: when hit some exts (or all fetch recxs hit) on replica space, the hi, lo, and iom_recxs are all valid, no?

File: src/object/cli_shard.c (hunk 4 above)
Comment (liuxuezhao) [reply src/object/cli_shard.c:394]: could be merge to one func with a flag to differentiate high/low, seems not quite necessary although.

File: src/object/cli_shard.c (hunk 3 above)
Comment (wangdi1) [reply src/object/cli_shard.c:374]: Hmm, I am confused here.  so this is trying to get the highest data cells?  then why skip it for the extent from parity node?

Comment (liuxuezhao) [reply src/object/cli_shard.c:374]: if it is from parity shard, then it is repilca ext that is in daos offset already right? so need not the converting.

Comment (wangdi1) [reply src/object/cli_shard.c:374]: I thought this is also choosing the max from all shards. (data or parity shard)

 hi.rx_idx = max(hi.rx_idx, rounddown(end - 1, cell_rec_nr));

No?

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**